queue/priority_queue.go

```diff
@@ -120,8 +120,9 @@ func (pq *PriorityQueue) Put(items ...Item) error {
 	}
 
 	pq.lock.Lock()
+	defer pq.lock.Unlock()
```

Comment:
I ran the benchmarks locally and got a similar result to what your benchmarks show. The defer version is actually faster for Put and Get. Interesting.

4-core i7 2.8 GHz, 16GB memory

```
[master][~/workspace/src/github.com/Workiva/go-datastructures/queue]$ go test -bench .
PASS
BenchmarkPriorityQueue-8         1000000              1618 ns/op
BenchmarkQueue-8         3000000               524 ns/op
BenchmarkChannel-8        500000              3726 ns/op
BenchmarkQueuePut-8        20000             84385 ns/op
BenchmarkQueueGet-8        20000             84042 ns/op
BenchmarkQueuePoll-8       20000             76694 ns/op
BenchmarkExecuteInParallel-8       30000             52584 ns/op
BenchmarkRBLifeCycle-8  20000000               111 ns/op
BenchmarkRBPut-8        20000000               112 ns/op
BenchmarkRBGet-8        30000000               247 ns/op
ok      github.com/Workiva/go-datastructures/queue      62.393s
[master][~/workspace/src/github.com/Workiva/go-datastructures/queue]$ gco jlhorona/master
Note: checking out 'jlhorona/master'.

You are in 'detached HEAD' state. You can look around, make experimental
changes and commit them, and you can discard any commits you make in this
state without impacting any branches by performing another checkout.

If you want to create a new branch to retain commits you create, you may
do so (now or later) by using -b with the checkout command again. Example:

  git checkout -b <new-branch-name>

HEAD is now at a942133... Priority queue cleanup
[a942133][~/workspace/src/github.com/Workiva/go-datastructures/queue]$ go test -bench .
PASS
BenchmarkPriorityQueue-8         1000000              1621 ns/op
BenchmarkQueue-8         3000000               503 ns/op
BenchmarkChannel-8        500000              3532 ns/op
BenchmarkQueuePut-8        20000             79595 ns/op
BenchmarkQueueGet-8        20000             81775 ns/op
BenchmarkQueuePoll-8       20000             78193 ns/op
BenchmarkExecuteInParallel-8       30000             60249 ns/op
BenchmarkRBLifeCycle-8  20000000               101 ns/op
BenchmarkRBPut-8        20000000                83.9 ns/op
BenchmarkRBGet-8        30000000               207 ns/op
ok      github.com/Workiva/go-datastructures/queue      59.569s
[a942133][~/workspace/src/github.com/Workiva/go-datastructures/queue]$
```